ibis/sql/mssql/client.py

```diff
@@ -0,0 +1,234 @@
+        if name == self.current_database or (
+            name is None and name != self.current_database
+        ):
+            return self.database_class(self.current_database, self)
+        else:
+            url = self.con.url
+            client_class = type(self)
+            new_client = client_class(
+                host=url.host,
+                user=url.username,
+                port=url.port,
+                password=url.password,
+                database=name,
+            )
+            return self.database_class(name, new_client)
+
+    def schema(self, name):
+        """Get a schema object from the current database for the schema named `name`.
+
+        Parameters
+        ----------
+        name : string
+
+        Returns
+        -------
+        schema : MSSQLSchema
+            An :class:`ibis.sql.mssql.client.MSSQLSchema` instance.
+
+        """
+        return self.database().schema(name)
+
+    @property
+    def current_database(self):
+        """Database client is currently connected to."""
+        return self.database_name
+
+    def list_databases(self):
+        """List all databases for client to connect to."""
+        return [
+            row.name
+            for row in self.con.execute(
+                'SELECT name FROM master.dbo.sysdatabases'
+            )
+        ]
+
+    def list_schemas(self):
+        """List all the schemas in the current database."""
+        return self.inspector.get_schema_names()
+
+    def set_database(self, name):
+        """Set current database that client is connected to."""
+        raise NotImplementedError(
+            'Cannot set database with MSSQL client. To use a different'
+            ' database, use client.database({!r})'.format(name)
+        )
+
+    @property
+    def client(self):
+        return self
+
+    def table(self, name, database=None, schema=None):
+        """Create an expression that references a particular table.
+
+        Parameters
+        ----------
+        name : string
+            The name of the table to retrieve.
+        database : string, optional
+            The database in which the table referred to by `name` resides. If
+            ``None`` then the ``current_database`` is used.
+        schema : string, optional
+            The schema in which the table resides.  If ``None`` then the
+            `public` schema is assumed.
+
+        Returns
+        -------
+        table : TableExpr
+            A table expression.
+        """
+        if database is not None and database != self.current_database:
+            return self.database(name=database).table(name=name, schema=schema)
+        else:
+            alch_table = self._get_sqla_table(name, schema=schema)
+            node = self.table_class(alch_table, self, self._schemas.get(name))
+            return self.table_expr_class(node)
+
+    def list_tables(self, like=None, database=None, schema=None):
+        """List tables avilable in current database."""
```

Comment:
@costrouc could you add to the docstrings the parameters and the return?

Reply:
added.